packages/table/src/table.tsx

```diff
@@ -810,6 +810,10 @@ export class Table extends AbstractComponent2<ITableProps, ITableState, ITableSn
             enableColumnInteractionBar,
         } = this.props;
         const { horizontalGuides, numFrozenColumnsClamped, numFrozenRowsClamped, verticalGuides } = this.state;
+        if (!this.gridDimensionsMatchProps()) {
```

Comment:
I feel like this should go into `componentDidUpdate` or another lifecycle method... I'm poking around at different places to move this check